src/QuokkaSimulation.hpp

```diff
@@ -1958,53 +1947,71 @@ auto QuokkaSimulation<problem_t>::computeHydroFluxes(amrex::MultiFab const &cons
 {
-	amrex::MultiArray4<const amrex::Real> x2State_fc_in;
-	amrex::MultiArray4<const amrex::Real> x3State_fc_in;
+	amrex::MultiArray4<const amrex::Real> x2State_fc_bfield_in;
+	amrex::MultiArray4<const amrex::Real> x3State_fc_bfield_in;
 	if constexpr (Physics_Traits<problem_t>::is_mhd_enabled) {
 		std::array<int, 3> delta_x2 = {0, 0, 0};
 		std::array<int, 3> delta_x3 = {0, 0, 0};
 		if constexpr (DIR == FluxDir::X1) {
 			delta_x2[1] = 1;
 			delta_x3[2] = 1;
-			x2State_fc_in = consVar_fc[1].const_arrays();
-			x3State_fc_in = consVar_fc[2].const_arrays();
+			x2State_fc_bfield_in = consVar_fc[1].const_arrays();
+			x3State_fc_bfield_in = consVar_fc[2].const_arrays();
 		} else if constexpr (DIR == FluxDir::X2) {
 			delta_x2[2] = 1;
 			delta_x3[0] = 1;
-			x2State_fc_in = consVar_fc[2].const_arrays();
-			x3State_fc_in = consVar_fc[0].const_arrays();
+			x2State_fc_bfield_in = consVar_fc[2].const_arrays();
+			x3State_fc_bfield_in = consVar_fc[0].const_arrays();
 		} else if constexpr (DIR == FluxDir::X3) {
 			delta_x2[0] = 1;
 			delta_x3[1] = 1;
-			x2State_fc_in = consVar_fc[0].const_arrays();
-			x3State_fc_in = consVar_fc[1].const_arrays();
+			x2State_fc_bfield_in = consVar_fc[0].const_arrays();
+			x3State_fc_bfield_in = consVar_fc[1].const_arrays();
 		}
-		auto primVar_in = primVar_mf.arrays();
+		auto bfield_cc_in = bfield_cc_mf.arrays();
 		amrex::IntVect ng{AMREX_D_DECL(nghost_fc_, nghost_fc_, nghost_fc_)};
-		amrex::ParallelFor(primVar_mf, ng, [=] AMREX_GPU_DEVICE(int bx, int i, int j, int k) {
-			const double bx2_m = x2State_fc_in[bx](i, j, k, Physics_Indices<problem_t>::mhdFirstIndex);
-			const double bx2_p = x2State_fc_in[bx](i + delta_x2[0], j + delta_x2[1], k + delta_x2[2], Physics_Indices<problem_t>::mhdFirstIndex);
-			primVar_in[bx](i, j, k, HydroSystem<problem_t>::x2Magnetic_index) = 0.5 * (bx2_m + bx2_p);
-
-			const double bx3_m = x3State_fc_in[bx](i, j, k, Physics_Indices<problem_t>::mhdFirstIndex);
-			const double bx3_p = x3State_fc_in[bx](i + delta_x3[0], j + delta_x3[1], k + delta_x3[2], Physics_Indices<problem_t>::mhdFirstIndex);
-			primVar_in[bx](i, j, k, HydroSystem<problem_t>::x3Magnetic_index) = 0.5 * (bx3_m + bx3_p);
+		amrex::ParallelFor(bfield_cc_mf, ng, [=] AMREX_GPU_DEVICE(int bx, int i, int j, int k) {
+			const double bx2_m = x2State_fc_bfield_in[bx](i, j, k, Physics_Indices<problem_t>::mhdFirstIndex);
+			const double bx2_p =
+			    x2State_fc_bfield_in[bx](i + delta_x2[0], j + delta_x2[1], k + delta_x2[2], Physics_Indices<problem_t>::mhdFirstIndex);
+			bfield_cc_in[bx](i, j, k, 0) = 0.5 * (bx2_m + bx2_p);
+
+			const double bx3_m = x3State_fc_bfield_in[bx](i, j, k, Physics_Indices<problem_t>::mhdFirstIndex);
+			const double bx3_p =
+			    x3State_fc_bfield_in[bx](i + delta_x3[0], j + delta_x3[1], k + delta_x3[2], Physics_Indices<problem_t>::mhdFirstIndex);
+			bfield_cc_in[bx](i, j, k, 1) = 0.5 * (bx3_m + bx3_p);
```

Comment:
It might help simplify things to extract this part of the code to a separate function, since it's duplicated here and in hydroFOFluxFunction.

Reply:
Yep, done.